suitcase/utils/__init__.py

```diff
@@ -27,6 +27,96 @@ class UnknownEventType(SuitcaseUtilsError):
     ...
 
 
+class Artifact:
+    """
+    A class that tracks information about a managed resource.
+
+    Parameters
+    ----------
+    label : string
+        A label for the sort of content being stored, such as
+        'stream_data' or 'metadata'.
+    postfix : string
+        Postfix for the file name. Must be unique per Manager.
+    name : string
+        Name of the file. The full file path or similar unique identifier.
+    handle: handle
+        A handle to a file, memory stream or similar.
+    """
+
+    def __init__(self, label, postfix, name=None, handle=None):
+        self.label = label
+        self.postfix = postfix
+        self.name = name
+
+        self.initial_size = None
+        self._final_size = None
+        self._handle = None
+
+        if handle is not None:
+            self.handle = handle
+
+    def __iter__(self):
+        """
+        Make this class iterable.
+
+        Allows converting instances to a dict by casting: dict(myartifact)
```

Comment:
Taking a fresh look at this, I wonder if we should just subclass `collections.Mapping` if we want this to be similar to a dict. Not worth slowing down this PR over it; it could always be added later.

Reply:
Yeah, I'm not a huge fan of this method of enabling converting this to a dictionary. It's not very direct, and the purpose of the code isn't obvious without that comment.

I remember considering subclassing but deciding against it... I think the main reasons were `handle` and `current_size` not fitting the `dict` paradigm very well. One being read-write and needing to be hooked on assignment, no matter how it is assigned, and the other being read-only and needing to be recalculated on access. Overall, the class isn't that `dict`like. 

The only reason I wanted to be able to convert it easily was so that `get_artifacts` can return the data in a plain format, so that `Artifact` can remain purely internal... and for some reason this was the best option that came to mind.

Thinking about it now, maybe a `to_dict` method would be more appropriate?

Reply:
Pushed the change... let me know if you think this is a good solution, or if subclassing Mapping would be better?

Reply:
I think that's a good solution. :+1: